src/arrlib.c

```diff
@@ -3,107 +3,248 @@
 #include <sys/types.h>
 #include <stdio.h>
 #include <string.h>
-
 #include "arrlib.h"
 
 typedef struct Node {
-    void *data1;
-    void *data2;
-    struct Node *next;
+	void *data1;
+	void *data2;
+	struct Node *next;
 } node_t;
 
 typedef struct Array {
-    struct Node *head;
-    size_t size_key;
-    size_t size_value;
-    size_t length;
+	struct Node *head;
+	struct Node *tail;
+	size_t size_key;
```

Comment:
just out of curiosity, what does this represent in our implementation? im assuming size_value is similar to array.length correct?

Reply:
we actually don't need it anymore-- good catch.